assets/shaders/fog.wgsl

```diff
@@ -0,0 +1,57 @@
+#import bevy_sprite::{mesh2d_view_bindings::globals}
+#import bevy_sprite::{mesh2d_view_bindings::view}
+#import bevy_sprite::mesh2d_vertex_output::VertexOutput
+#import "shaders/perlin_noise_3d.wgsl"::perlinNoise3
+
+struct FogMaterial {
+    depth: f32,
+    alpha: f32,
+    color: vec4<f32>,
+    emitter1: vec4<f32>,
+}
+
+@group(1) @binding(0) var<uniform> fog_mat: FogMaterial;
+
+@fragment
+fn fragment(mesh: VertexOutput) -> @location(0) vec4<f32> {
+    var world_pos = mesh.world_position.xy;
+    let camera_pos = view.world_position;
+
+    // calculates parallax offset
+    // works by scaling the effective distance of the emitter from the camera
+    let emitter_pos_init = fog_mat.emitter1.xy;
+    var delta = emitter_pos_init - camera_pos.xy;
+    delta *= 1.0/ (fog_mat.depth);
+    let emitter_pos_final = camera_pos.xy + delta.xy;
+
+    // offset is needed so that world_space moves with the parallax
+    // this way fog won't "roll" whenever the camera moves.
+    let offset = emitter_pos_final - emitter_pos_init;
+    let scaledpos = (world_pos - offset)*0.05;
+
+    // low frequency noise
+    var low_f_noise = perlinNoise3(vec3<f32>(scaledpos.x*0.32, scaledpos.y*0.32, globals.time*0.1 + 1.0 / fog_mat.depth));
+
+    // Generates cool noise affect by combining squared noise with subtracted, different scaled noise.
+    // Note that we use time as the z coordinate, and offset it by depth; this makes the fog
+    // look 3d; different fogs in different layers will be different slices of the same 3d noise
+    // resulting in a better effect.
+    var noise = perlinNoise3(vec3<f32>(scaledpos.x, scaledpos.y, globals.time*0.25 + 10.0 / fog_mat.depth));
+    var noise1 = perlinNoise3(vec3<f32>(scaledpos.x*0.5, scaledpos.y*0.5, globals.time*0.25 + 5.0 / fog_mat.depth));
```

Comment:
Ideally we should precompute the noise and sample it from a texture, instead of generating it in real time in the fragment shader every time?

It's up to you whether to do that in Rust (btw there is a `noise` crate we could add as a dependency if it would help) or whether you want to use your existing gpu-based shader noise computation function (we could move it to a compute shader that gets run once to generate the texture).